### ISSUE TITLE
A little suggestion about "if let"

### ISSUE THREAD

**Issue by roadhoghook** [rust-lang/rust#68250]
The result of this code is  " three!! " , I think rustc should warn about the wrong line .
I am a beginner. I hope you can correct me if something i did wrong.
```rust
    let wrong_doing = Some(2);
    if let wrong_doing = Some(3) { // if let Some(3) = wrong_doing
       println!("Three!!");
    } else {
        println!("Not Three");
    }
```

**Comment by jonas-schievink** [comment]
It already warns:

```
warning: irrefutable if-let pattern
 --> src/main.rs:3:5
  |
3 | /     if let wrong_doing = Some(3) { // if let Some(3) = wrong_doing
4 | |        println!("Three!!");
5 | |     } else {
6 | |         println!("Not Three");
7 | |     }
  | |_____^
```
